pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -358,6 +358,19 @@ public synchronized PinotResourceManagerResponse addInstance(Instance instance)
     }
   }
 
+  /**
+   * Update a given instance for the specified Instance ID
+   */
+  public synchronized PinotResourceManagerResponse updateInstance(String instanceIdToUpdate, Instance newInstance) {
+    List<String> instances = getAllInstances();
+    if (!instances.contains(instanceIdToUpdate)) {
+      return PinotResourceManagerResponse.failure("Instance " + instanceIdToUpdate + " does not exists");
+    } else {
+      _helixAdmin.setInstanceConfig(_helixClusterName, instanceIdToUpdate, newInstance.toInstanceConfig());
```

Comment:
This won't be able the change the instance host/port. In order to do that, you need to use data accessor to set it.